packages/strapi-plugin-i18n/services/localizations.js

```diff
@@ -24,8 +48,27 @@ const updateNonLocalizedFields = async (entry, { model }) => {
   }
 };
 
+/**
+ * Remove entry from localizations & udpate realted localizations
+ * @param {Object} entry entry to remove from localizations
+ * @param {Object} options
+ * @param {Object} options.model corresponding model
+ */
+const removeEntryFromLocalizations = async (entry, { model }) => {
+  if (Array.isArray(entry.localizations)) {
+    const newLocalizations = entry.localizations.filter(({ id }) => id != entry.id);
```

Comment:
Don't we need to remove the entry from its own localization attribute too? Or do we assume that if we remove an entry from a localization object, it's because the entry has been deleted or we removed i18n from it?

Reply:
This function is only called after deletion so I can't update the entry. Let's just assume it will be deleted already for now.